dask_kubernetes/core.py

```diff
@@ -14,19 +17,186 @@
+class Pod(ProcessInterface):
+    """ A superclass for Kubernetes Pods
+    See Also
+    --------
+    Worker
+    Scheduler
+    """
+
+    def __init__(self, core_api, pod_template, namespace, loop=None, **kwargs):
+        self.pod = None
+        self.core_api = core_api
+        self.pod_template = copy.deepcopy(pod_template)
+        self.namespace = namespace
+        self.name = None
+        self.loop = loop
+        self.kwargs = kwargs
+        super().__init__()
+
+    @property
+    def cluster_name(self):
+        return self.pod_template.metadata.labels["dask.org/cluster-name"]
+
+    async def start(self, **kwargs):
+        self.pod = await self.core_api.create_namespaced_pod(
+            self.namespace, self.pod_template
+        )
+
+        while self.pod.status.phase == "Pending":
+            self.pod = await self.core_api.read_namespaced_pod(
+                self.pod.metadata.name, self.namespace
+            )
+            await asyncio.sleep(0.1)
+
+        while self.address is None:
+            logs = await self.logs()
+            for line in logs.splitlines():
+                for query_string in ["worker at:", "Scheduler at:"]:
+                    if query_string in line:
+                        self.address = line.split(query_string)[1].strip()
+            await asyncio.sleep(0.1)
+
+        await super().start(**kwargs)
+
+    async def close(self, **kwargs):
+        if self.pod:
+            await self.core_api.delete_namespaced_pod(
+                self.pod.metadata.name, self.namespace
+            )
+        await super().close(**kwargs)
+
+    async def logs(self):
+        return Log(
+            await self.core_api.read_namespaced_pod_log(
+                self.pod.metadata.name, self.namespace
+            )
+        )
+
+    def __repr__(self):
+        return "<Pod %s: status=%s>" % (type(self).__name__, self.status)
+
+
+class Worker(Pod):
+    """ A Remote Dask Worker controled by SSH
+    Parameters
+    ----------
+    scheduler: str
+        The address of the scheduler
+    address: str
+        The hostname where we should run this worker
+    connect_kwargs: dict
+        kwargs to be passed to asyncssh connections
+    kwargs:
+        TODO Document Worker kwargs
+    """
+
+    def __init__(self, scheduler: str, **kwargs):
+        super().__init__(**kwargs)
+
+        self.scheduler = scheduler
+
+        self.pod_template.metadata.labels["dask.org/component"] = "worker"
+        self.pod_template.spec.containers[0].env.append(
+            kubernetes.client.V1EnvVar(
+                name="DASK_SCHEDULER_ADDRESS", value=self.scheduler
+            )
+        )
+
+
+class Scheduler(Pod):
+    """ A Remote Dask Scheduler controled by SSH
+    Parameters
+    ----------
+    address: str
+        The hostname where we should run this worker
+    connect_kwargs: dict
+        kwargs to be passed to asyncssh connections
+    kwargs:
+        TODO Document Scheduler kwargs
```

Comment:
Can we defer users to the `dask.distributed.Scheduler` docstring?

Reply:
Is there a neat way of doing this in Python? Or should I just provide a link to the dask docs?

Reply:
Probably, but I would just link, and maybe add an entry in `See Also`